vms/platformvm/service.go

```diff
@@ -1281,10 +1278,17 @@ func (s *Service) AddValidator(req *http.Request, args *AddValidatorArgs, reply
 
 	reply.TxID = tx.ID()
 	reply.ChangeAddr, err = s.addrManager.FormatLocalAddress(changeAddr)
+	if err != nil {
+		return fmt.Errorf("could not format address %s: %w", changeAddr, err)
+	}
+
+	s.vm.ctx.Lock.Unlock()
+	err = s.vm.Network.IssueTx(req.Context(), tx)
+	s.vm.ctx.Lock.Lock()
+	defer s.vm.ctx.Lock.Unlock()
```

Comment:
While I may permit this kind of stuff in tests... I don't feel comfortable merging locking behavior like this (unless absolutely necessary). I think these changes would be less error prone if they were more similar to the changes here: #2524